### PR TITLE
Feature/aggregate operations order time stateful

### PR DESCRIPTION
**Pull request by gserranojc**

### CODE REVIEW

File: examples/data-generators/ecommerce/ecommerce-policy.json [hunk 1]
```diff
@@ -2,6 +2,11 @@
   "name": "ecommerce-policy",
   "duration": 10000,
   "saveRawData": "false",
+  "checkpointDir": "checkpoint",
+  "timeBucket": "minute",
+  "checkpointGranularity": "minute",
```
Comment (sgomezg) [inline examples/data-generators/ecommerce/ecommerce-policy.json]: I don't understand this change. Did you mean "30000 minutes"?

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala [hunk 2]
```diff
@@ -22,11 +22,24 @@ import java.util.Date
...
+  final val TIMESTAMP_PATTERN = "@timestamp:"
+  final val DEFAULT_DATE_FORMAT = "YYYY.MM.dd"
+  final val YEAR = "YYYY"
+  final val MONTH = "MM"
+  final val DAY = "dd"
+  final val HOUR = "HH"
+  final val MINUTE = "mm"
+  final val SECOND = "ss"
+  final val DEFAULT_INDEX_TYPE = "sparkta"
+  final val DEFAULT_NODE = "localhost"
```
Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: This is a bad idea. We doesn't want a default config in code. Moreover, this config only works in local machine!

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala [hunk 3]
```diff
@@ -22,11 +22,24 @@ import java.util.Date
...
+import com.stratio.sparkta.sdk.TypeOp
 import com.stratio.sparkta.sdk.TypeOp._
-import com.stratio.sparkta.sdk.{Output, TypeOp}
+import com.stratio.sparkta.sdk._
 
 trait ElasticSearchDAO extends Closeable {
 
+  final val TIMESTAMP_PATTERN = "@timestamp:"
+  final val DEFAULT_DATE_FORMAT = "YYYY.MM.dd"
+  final val YEAR = "YYYY"
```
Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: Maybe we need a constant object for reusing this values.

**Comment by sgomezg** [comment]
You have to update doc due to policy changes.

**Comment by sgomezg** [comment]
PR description is a must :)

File: examples/data-generators/ecommerce/ecommerce-policy.json (hunk 1 above)
Comment (gserranojc) [inline examples/data-generators/ecommerce/ecommerce-policy.json]: This value is milliseconds. Is the additional time that you want extend the updateStateByKey.

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala (hunk 2 above)
Comment (gserranojc) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: In other PR we discuss this, they are default parameters.

Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: I remember that. Default config for Elasticsearch pointing to localhost is a BAD idea. This is only for dev mode, almost exclusively for dev environment. This wouldn't be default case.

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala (hunk 3 above)
Comment (gserranojc) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: These constants are elasticsearch plugin, do not have to use other modules of the application.

Comment (gserranojc) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: We need to update many contents of the documentation, we get into another PR. In addition to expanding the content to explaining all the new functionality.
If you notice there are very important things that they have changed drastically.

Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDAO.scala]: Sorry but I'm not agree. If you modify something important, documentation must go with the modifications.

File: examples/data-generators/ecommerce/ecommerce-policy.json (hunk 1 above)
Comment (gserranojc) [inline examples/data-generators/ecommerce/ecommerce-policy.json]: I documented all in the section usingSparkta.

**Comment by sgomezg** [comment]
Please rebase to correct conflicts and it will be ok to me :)

**Pull request merged**